### ISSUE TITLE
 vscode.workspace.openTunnel() opens tunnel with different domain in Github Codespaces

### ISSUE THREAD

**Issue by colombod** [microsoft/vscode#126304]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.58-Insider
- OS Version:

In codespaces .NET Interactive extension calls openTunnel() to expose ports.
As you can see in screenshot the codepsaces is on domain `.github.dev` but the tunnel `.githubpreview.dev` that giving errors for COR
![image](https://user-images.githubusercontent.com/375556/121965574-2456ee80-cd65-11eb-95d8-762d5ca1c941.png)

![image](https://user-images.githubusercontent.com/375556/121965401-e063e980-cd64-11eb-8c7a-f0f2c494cb25.png)

**Comment by alexr00** [comment]
@cmbrose, this is expected, correct?

**Comment by colombod** [comment]
> @cmbrose, this is expected, correct?

If this is expected what steps should we do to ensure that is not a COR violation? we load js and other static content from that tunnel url.

**Comment by alexr00** [comment]
I meant to ask @cmbrose if the different domain was expected, not if a COR violation is expected :)

@joaomoreno I think you understand COR issues. Do you have any ideas for how to investigate this?

**Comment by cmbrose** [comment]
>@cmbrose, this is expected, correct?

Yes, this is by design. For codespaces use customer apps will need to be able to handle this in their COR policies

**Comment by colombod** [comment]
> > @cmbrose, this is expected, correct?
> 
> Yes, this is by design. For codespaces use customer apps will need to be able to handle this in their COR policies

Do you have any pointer we can look at? we are configuring our fetch (used in webview) to use the uri from tunnel to access the server on the back end. How can we configure it to allow that domain too?

**Comment by colombod** [comment]
@brettfo

**Comment by cmbrose** [comment]
I unfortunately don't know very much about how the webview works as that is entirely within vscode - @alexr00 could you point us to the expert there?

**Comment by alexr00** [comment]
@mjbvz is the webview expert. Closing as there is no code action that needs to be take for this issue.
